sopel/bot.py

```diff
@@ -179,6 +177,23 @@ def write(self, args, text=None):  # Shim this in here for autodocs
         irc.Bot.write(self, args, text=text)
 
     def setup(self):
+        """Setup Sopel bot before it can run
+
+        The setup phase manages to:
+
+        * setup logging (configure Python's built-in :mod:`logging`),
+        * setup the bot's plugins (load, setup, and register)
+        * start the job scheduler
+
+        """
+        self.setup_logging()
+        self.setup_plugins()
+        self.scheduler.start()
+
+    def setup_logging(self):
+        logger.setup_logging(self)
+
+    def setup_plugins(self):
```

Comment:
I can't add a line note below because GitHub arbitrary limitations, but I don't think this function should print "Welcome to Sopel." It should only print "Loading plugins..." (not "modules"), and any errors encountered while doing that.

A "Welcome" message should be printed elsewhere, if at all. Maybe at the beginning of `setup()`? Assuming that would work properly before `setup_logging()` runs.

Reply:
Yeah, I'd rather see no message at all. I think a rework of logging could be a great thing to consider. I'll remove that for now.

Reply:
After further thought, I really want to keep some kind of "welcome" message. But the one I envision includes version numbers for Sopel itself and the Python runtime, and that's an enhancement for a separate PR IMO.